flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -258,30 +268,37 @@ private SnapshotDirectory prepareLocalSnapshotDirectory(long checkpointId) throw
         }
```

Comment:
Aborted **files** can be re-used, but not `StateHandleID`s held by `materializedSstFiles` (because JM might not remember these keys on abortion).

> edit: actually I think the logic maybe is correct, but something feels inconsistent here. It's really hard for me to name what is inside materializedSstFiles. Those are confirmed files and also those awaiting confirmation. But aborted files are being removed from it.

Right. This logic is the same as before. The purpose of `notifyCheckpointAborted` is purely cleanup. It's called from state backend, so it's not dead code.

The purpose of `lastUploadedSstFiles` is not to handle abortion:
- if a file was not confirmed for any reason, then `StreamStateHandle` must be sent
- if it was, then a placeholder can be sent

Reply:
>  The purpose of notifyCheckpointAborted is purely cleanup. It's called from state backend, so it's not dead code.

In this context by dead code I meant "not having any meaningful effect". The code would be working just as well without this `notifyCheckpointAborted()`, right? The `materializedSstFiles` would still be cleaned up in `notifyCheckpointComplete()`?

About my proposal of unifying `lastUploadedSstFiles` and `materializedSstFiles` I missed that the former one has also `StreamStateHandle`s. So to unify them, we would need to change `materializedSstFiles` type to: `SortedMap<Long, Map<StateHandleID, StreamStateHandle>>`. (so keeping them separate makes sense).

However I'm still confused by those variables and the inconsistency between them. Once checkpoint is aborted, those sst files are no longer "materialized"? But they are still considered as uploaded? At the very least naming is confusing.

Reply:
> In this context by dead code I meant "not having any meaningful effect". The code would be working just as well without this notifyCheckpointAborted(), right? The materializedSstFiles would still be cleaned up in notifyCheckpointComplete()?

Got you. Yes, correctness won't be affected by removing this method.

I agree that the naming is confusing. `materializedSstFiles` should probably be called `uploadedStateIDs`. `lastUploadedSstFiles` can then be called `uploadedStateHandles` (I've done this in 144aa68718e831cf07c2daa60b2d33ae149ce592).
WDYT?

I've also tried to combine the two maps into a single `SortedMap<Long, Snapshot> snapshotsByCheckpoint`. But that didn't work because on checkpoint, we might need data from potentially two checkpoints: last uploaded and last confirmed.

Reply:
👍